src/main/java/org/folio/services/transactions/CommonTransactionService.java

```diff
@@ -103,6 +103,25 @@ public Future<Void> releaseTransaction(Transaction transaction, RequestContext r
                     .compose(summary -> updateTransaction(transaction, requestContext));
   }
 
+  public Future<Void> unreleaseTransaction(String id, RequestContext requestContext) {
+    return retrieveTransactionById(id, requestContext)
+      .compose(transaction -> unreleaseTransaction(transaction, requestContext));
+  }
+
+  public Future<Void> unreleaseTransaction(Transaction transaction, RequestContext requestContext) {
+    logger.info("Start unreleasing transaction {}", transaction.getId()) ;
+
+    validateTransactionType(transaction, Transaction.TransactionType.ENCUMBRANCE);
+
+    if (transaction.getEncumbrance().getStatus() != Encumbrance.Status.RELEASED) {
+      return succeededFuture(null);
```

Comment:
Maybe we need to fail because it's called in `POST` method

Reply:
we need to support consistency with contract of release endpoint, if we going to do change - need to apply the same in that endpoint as well, but as we for me can remain as is